common/tools/dev-tool/src/commands/run/extract-api.ts

```diff
@@ -83,30 +88,50 @@ export default leafCommand(commandInfo, async () => {
   // sub path exports extraction
   const exports = packageJson["exports"];
   if (exports) {
-    for (const exprt of Object.keys(exports)) {
-      log.debug(`extracting Api for "${exprt}"`);
-      const suffix = getSuffix(exprt);
-      const newPublicTrimmedPath = extractorConfigObject.dtsRollup.publicTrimmedFilePath.replace(
-        ".d.ts",
-        `-${suffix}.d.ts`,
-      );
-      const newApiJsonPath = extractorConfigObject.docModel?.apiJsonFilePath?.replace(
-        ".api.json",
-        `-${suffix}.api.json`,
-      );
-      const newApiReportName = extractorConfigObject.apiReport?.reportFileName?.replace(
-        ".api.md",
-        `-${suffix}.api.md`,
+    log.info("Detected subpath exports, extracting markdown for each subpath.");
+    for (const exportPath of Object.keys(exports)) {
```

Comment:
You should also omit `package.json` since it is by default in `tshy`. See [core-sse package.json](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-util/package.json#L10)

Reply:
Interesting, thanks! That reminds me that tshy aware packages look much different than our current setup and may not work with this subpath config at all... I suspect we'll need to update the logic here once more packages adopt tshy. 

With that in mind, I am actually inclined to tackle that separately from this PR and will make an issue to handle core packages (and in the process tackle the different package.json shapes)